### ISSUE TITLE
getCellMeta creates metadata for non-existing cells

### ISSUE THREAD

**Issue by ottosson** [handsontable/handsontable#7338]
### Description
If I use getCellMeta for a cell that does not exist, it seems to be created instead of returning null.

### Steps to reproduce

1. Use hot.getCellMeta() for a cell that does not exist
2. Use hot.getCellsMeta() to get all table metadata
3. See that now you have table cell count + 1 results

### Demo
https://jsfiddle.net/mbzvck9q/

### Your environment
* Handsontable version: 8.1.0
* Browser Name and version: latest Chrome
* Operating System: WIndows 10

**Comment by AMBudnik** [comment]
Hi @ottosson I replicated the same behavior. It seems that when we call a cell that is distant (not a next index) we do not get all the cells between, just the new one

```js
data: Handsontable.helper.createSpreadsheetData(10, 6),
...
const y = hot.getCellMeta(20, 0);
```
![Zrzut ekranu 2020-10-15 o 09 23 07](https://user-images.githubusercontent.com/10757813/96090298-48a5b700-0ec8-11eb-874f-99676713dabe.png)

However, this behavior changed in v 8.0.0 https://jsfiddle.net/AMBudnik/ums2p8kv/ as in 7.4.2 https://jsfiddle.net/AMBudnik/g7p6mhyq/ we would get all the cell meta values, which is 71 records.

But the question is, s**hould we allow receiving anything besides undefined for calling cell meta for cells that do not exist**? We'd need to discuss the subject.

**Comment by ottosson** [comment]
If the cell does not exist, why get anything at all? Possibly throw an error?

**Comment by AMBudnik** [comment]
> If the cell does not exist, why get anything at all? Possibly throw an error?

Yes. That is what I would expect. However, I remember having a similar discussion before with the team and opinions were mixed. So I wanted to attach the **Status: To be discussed** label instead of the **bug**.

**Comment by ottosson** [comment]
Yup, got it. I just wanted to add my 2 cents on the matter 🙂

**Comment by budnix** [comment]
This is expected behavior.

`getCellMeta()` does not return `null` for cells outside the current dataset because Handsontable can theoretically expand the dataset indefinitely, both by rows and columns. Because of that, access to cell meta is technically always available.

For cells that don’t exist in the source data, the returned cell meta object is still built from the table-level settings, following Handsontable’s cascading configuration model.

Closing this one as by design.
